lib/trailblazer/rails/controller.rb

```diff
@@ -10,9 +10,15 @@ def run_v21(operation, *dependencies, **variables)
 
       _assign_trb_ivars(result)
 
-      yield(result) if result.success? && block_given?
-
       @_result = result
+
+      return unless result.success? && block_given?
```

Comment:
This should be `return if result.success?` as we're only interested in this case, right?